db/memtable.cc

```diff
@@ -965,6 +955,21 @@ static bool SaveValue(void* arg, const char* entry) {
   if (user_comparator->EqualWithoutTimestamp(user_key_slice,
                                              s->key->user_key())) {
     // Correct user key
+    std::unique_ptr<ReadLock> read_lock;
+    if (s->inplace_update_support) {
+      read_lock.reset(new ReadLock(s->mem->GetLock(s->key->user_key())));
+    }
+
+    if (s->protection_bytes_per_key > 0) {
+      *(s->status) = MemTable::VerifyEntryChecksum(
+          entry, s->protection_bytes_per_key, s->allow_data_in_errors);
+      if (!s->status->ok()) {
+        ROCKS_LOG_ERROR(s->logger, "In SaveValue: %s", s->status->getState());
+        // Memtable entry corrupted
+        return false;
+      }
+    }
```

Comment:
fyi @cbi42 I'm thinking about reducing the `SaveValue()` memtable verification to just the matching keys. Alternatively I could reduce the verification only when `inplace_update_support == true` if that's preferred.

Reply:
That's fine. For non-matching keys' verification, they can be included later as part of verification during look up.